### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

**Comment by harrism** [comment]
This PR needs a description.

File: .github/labeler.yml
Comment (mike-wendt) [inline .github/labeler.yml:17]: This section should match this repo's labels and the contents of the .github/CODEOWNERS file as a starting guide - PICs in this repo would then add or suggest more labels as needed

File: .github/labeler.yml
Comment (mike-wendt) [inline .github/labeler.yml:25]: This section should be removed

**Review by mike-wendt — CHANGES_REQUESTED** [review]

File: .github/labeler.yml
Comment (ajschmidt8) [inline .github/labeler.yml:16]:
```suggestion
gpuCI:
```
There's no `ci` label, so this can probably be changed to `gpuCI`.

File: .github/labeler.yml
Comment (ajschmidt8) [inline .github/labeler.yml:9]: There is no `cpp` label.

Comment (jolorunyomi) [reply .github/labeler.yml:9]: do you know what I can replace cpp with?

**Comment by thomcom** [comment]
@mike-wendt mind updating your review when you get the chance?

File: .github/labeler.yml
Comment (ajschmidt8) [reply .github/labeler.yml:9]: probably fine to leave this. based on Mike's comment in the other PRs, this label will be created if it doesn't already exist.

**Review by mike-wendt — APPROVED** [review]
Approved, but defer to @trxcllnt and @thomcom for final approval as they are the most active committers on this repo. Feel free to suggest or change labels/paths

**Comment by mike-wendt** [comment]
@trxcllnt @thomcom feel free to update the config if needed, going to merge this as-is to get started

**Comment by mike-wendt** [comment]
@gpucibot merge

**Pull request merged**
